viewing_party/party.py

```diff
@@ -0,0 +1,204 @@
+def valid_str(inp):
+    ''' 
+    check for valid string imput
+    '''
+    # is it a string
+    str_stat = isinstance(inp, str)
+    # is it nonempty
+    str_exists = bool(inp)
+    return str_stat and str_exists
+
+def valid_rating(rating):
+    '''
+    check for valid numeric input (notably for rating)
+    '''
+    # is it a number
+    rating_stat = isinstance(rating, (int, float))
+    if rating_stat:
+        # is it in range
+        if 0 <= rating <= 5:
+            return True
+        else:
+            return False
+    return False # not a number
+
+def create_movie(title, genre, rating):
+    '''
+    if valid inputs for title, genre, & rating, return movie dict
+    else: return None
+    '''
+    if valid_str(title) and valid_str(genre) and valid_rating(rating):
+        # build dict
+        movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie
+    else:
+        return None
+
+
+def add_to_watched(user_data, movie):
+    '''
+    the value of user_data will be a dictionary with a key 
+    "watched", and a value which is a list of dictionaries 
+    representing the movies the user has watched
+    add the movie to the "watched" list inside of user_data
+    '''
+    # verify 'watched' is a key in user_data
+    user_data["watched"].append(movie)
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    '''
+    the value of user_data will be a dictionary with a 
+    key "watchlist", and a value which is a list of 
+    dictionaries representing the movies the user wants to watch
+    & add the movie to the "watchlist" list inside of user_data
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    user_data: dictionary with a "watchlist" and a "watched"
+    title: string that represents the title of the movie 
+        the user has watched
+    if movie in watchlist, remove from watchlist, add to watched
+    returns user_data
+    '''
+    for movie in user_data["watchlist"]: 
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    num_movies = len(user_data["watched"])
+    if num_movies > 0:
+        sum = 0
+        for movie in user_data["watched"]:
+            sum += movie["rating"]
+        avg = sum / num_movies
+    else:
+        avg = 0.0
+    return avg
+
+
+def get_most_watched_genre(user_data): 
+    '''
+    determines which genre is most frequently occurring
+    return most frequently watched
+    if not applicable, return None
+    '''
+    # if there are watched movies, count genre freq
+    if len(user_data["watched"])> 0:
+        # store counters in genres
+        genres = dict()
```

Comment:
you know, I'm not sure what's under the hood of this class instantiation, but it might save time by simply creating the empty dic yourself:

```suggestion
        genres = {}
```

it's a very minimal change, may not make much of a difference efficiency wise, but it _does_ stop you from calling a function